### ISSUE TITLE
nodata issue

### ISSUE THREAD

**Issue by giswqs** [banesullivan/localtileserver#58]
It seems the `nodata` parameter has no effect. No matter what `nodata` value is provided (e.g., -1), the tile will always treat 0 as nodata, resulting in some empty areas in the image.  

```python
from localtileserver import get_leaflet_tile_layer, TileClient
from ipyleaflet import Map

# First, create a tile server from local raster file
tile_client = TileClient('https://github.com/giswqs/data/raw/main/raster/landsat7.tif')

# Create ipyleaflet tile layer from that server
t = get_leaflet_tile_layer(tile_client, band=[1,2,3], nodata=-1)

# Create ipyleaflet map, add tile layer, and display
m = Map(center=tile_client.center())
m.add_layer(t)
m
```
![image](https://user-images.githubusercontent.com/5016453/148670352-685f9ab8-243f-4541-aeaa-9a3a9bf1b9b5.png)

![image](https://user-images.githubusercontent.com/5016453/148670362-997f656f-66af-4029-85e7-407cace28b71.png)

**Comment by banesullivan** [comment]
Huh. this surprises me... I will investigate

**Comment by banesullivan** [comment]
I think this might be an upstream issue: https://github.com/girder/large_image/issues/756

**Comment by giswqs** [comment]
Thanks for looking into it.

**Comment by giswqs** [comment]
Fixed in #60

**Comment by giswqs** [comment]
This issue is happening again. The `nodata` param has not effect. 

```python
from localtileserver import get_leaflet_tile_layer, TileClient
from ipyleaflet import Map

# First, create a tile server from local raster file
tile_client = TileClient('https://github.com/giswqs/data/raw/main/raster/landsat7.tif')

# Create ipyleaflet tile layer from that server
t = get_leaflet_tile_layer(tile_client, band=[1,2,3], nodata=-1)

# Create ipyleaflet map, add tile layer, and display
m = Map(center=tile_client.center())
m.add_layer(t)
m
```
![image](https://user-images.githubusercontent.com/5016453/148670352-685f9ab8-243f-4541-aeaa-9a3a9bf1b9b5.png)

![image](https://github.com/banesullivan/localtileserver/assets/5016453/b9ac889c-fb95-42a3-8b77-f76347d12a31)

**Comment by banesullivan** [comment]
I'm about to issue a new release that is entirely based on rio-tiler. I was looking into this and it seems this may be an upstream issue:

```py
from rio_tiler.io import Reader

class Im(bytes):
    def _repr_png_(self):
        return self

reader = Reader('https://github.com/giswqs/data/raw/main/raster/landsat7.tif')

Im(reader.preview(nodata=-1, max_size=256).render())
```

I will raise in rio-tiler

**Comment by banesullivan** [comment]
Confirmed fixed upstream with rio-tiler 6.3.1 and 0.10.0rc2 localtileserver
